snuba/clickhouse/query.py

```diff
@@ -0,0 +1,90 @@
+from typing import Sequence
+
+from snuba import util
+from snuba.datasets import Dataset
+from snuba.schemas import Request
+
+
+class ClickhouseQuery:
+    def __init__(self,
+        dataset: Dataset,
+        # TODO: replace this with snuba.query. Query as soon as the PR is committed
+        request: Request,
+        prewhere_conditions: Sequence[str],
+        final: bool,
+    ) -> None:
+        self.__dataset = dataset
+        self.__request = request
+        self.__prewhere_conditions = prewhere_conditions
+        self.__final = final
+
+    def format(self) -> str:
```

Comment:
daaaaaamn, merging #456 into this will be a nightmare. Did not think about this

Reply:
#460 too

Reply:
Meh #460 should be fine. It's a pretty easy change anyway.